src/main/java/edu/ucsb/cs156/happiercows/controllers/ProfitsController.java

```diff
@@ -0,0 +1,209 @@
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/admin/post")
+    public Profit postProfit_admin(
+            @ApiParam("profit") @RequestParam long profit,
+            @ApiParam("timestamp") @RequestParam long timestamp,
+            @ApiParam("userCommonsId") @RequestParam long userCommonsId) {
+        UserCommons userCommons = userCommonsRepository.findById(userCommonsId).orElseThrow(() -> new EntityNotFoundException(UserCommons.class, userCommonsId));
+
+        Profit createdProfit = new Profit();
+        createdProfit.setProfit(profit);
+        createdProfit.setUserCommons(userCommons);
+        createdProfit.setTimestamp(timestamp);
+        Profit savedProfit = profitRepository.save(createdProfit);
+        return savedProfit;
+    }
+
+    @ApiOperation(value = "Create a new Profit as user")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("/post")
+    public Profit postProfit(
+            @ApiParam("profit") @RequestParam long profit,
+            @ApiParam("timestamp") @RequestParam long timestamp,
+            @ApiParam("userCommonsId") @RequestParam long userCommonsId) {
+
+        Long userId = getCurrentUser().getUser().getId();
+        UserCommons userCommons = userCommonsRepository.findById(userCommonsId).orElseThrow(() -> new EntityNotFoundException(UserCommons.class, userCommonsId));
+
+        Profit createdProfit = new Profit();
+        createdProfit.setProfit(profit);
+        createdProfit.setUserCommons(userCommons);
+        createdProfit.setTimestamp(timestamp);
+        Profit savedProfit = profitRepository.save(createdProfit);
+        return savedProfit;
+    }
+
+    @ApiOperation(value = "Update a single profit as admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/admin")
+    public Profit putProfitById_admin(
+            @ApiParam("id") @RequestParam Long id,
+            @RequestBody @Valid Profit newProfit) {
+        Profit profit = profitRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Profit.class, id));
+
+        profit.setProfit(newProfit.getProfit());
+        profit.setTimestamp(newProfit.getTimestamp());
+
+        profitRepository.save(profit);
+
+        return profit;
+    }
+
+    @ApiOperation(value = "Update a single profit as user")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping("")
+    public Profit putProfitById(
+            @ApiParam("id") @RequestParam Long id,
+            @RequestBody @Valid Profit newProfit) {
+        Long userId = getCurrentUser().getUser().getId();
+
+        Profit profit = profitRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Profit.class, id));
+        if (userId != profit.getUserCommons().getUserId())
+            throw new EntityNotFoundException(Profit.class, id);
+
+        profit.setProfit(newProfit.getProfit());
+        profit.setTimestamp(newProfit.getTimestamp());
+
+        profitRepository.save(profit);
+
+        return profit;
+    }
+

```

Comment:
What is the use case for a user to update their own profit?

This seems like a way that a user could cheat and just give themselves more money.